src/bicep-types/src/types.ts

```diff
@@ -163,6 +161,8 @@ export type ResourceType = TypeBase<TypeBaseKind.ResourceType, {
   body: TypeReference;
   flags: ResourceFlags;
   functions?: Record<string, ResourceTypeFunction>;
+  readableScopes?: ScopeType;
+  writableScopes?: ScopeType;
 }>
```

Comment:
`bicep-types` is just for writing the types.json format, right? I would make the same changes here as to the C# ResourceType